netbsd/NetBSDProcessList.c

```diff
@@ -361,12 +378,58 @@ static void NetBSDProcessList_scanCPUTime(NetBSDProcessList* this) {
    kernelCPUTimesToHtop(avg, this->cpus);
 }
 
+static void NetBSDProcessList_scanCPUFrequency(NetBSDProcessList* this) {
+   unsigned int cpus = this->super.cpuCount;
+   bool match = false;
+   char name[64];
+   int freq = 0;
+   size_t freqSize = sizeof(freq);
+
+   for (unsigned int i = 0; i <= cpus; i++) {
+         this->cpus[i].frequency = NAN;
+   }
+
+   /* newer hardware supports per-core frequency, for e.g. ARM big.LITTLE */
+   for (unsigned int i = 0; i <= cpus; i++) {
+      snprintf(name, sizeof(name), "machdep.cpufreq.cpu%u.current", i);
+      if (sysctlbyname(name, &freq, &freqSize, NULL, 0) != -1) {
```

Comment:
Doesn't need `freqsize` a re-initialization before each call?

Reply:
Hmm, maybe, I doubt the kernel will ever return a different size than the initial oldlenp. But we can change it just to be safe.